Problem096/C++/solution_1.cpp

```diff
@@ -0,0 +1,154 @@
+#include <iostream>
+#include<fstream>
+#include<stdlib.h>
+#include<stdio.h>
+#include<time.h>
+#define UNASSIGNED 0
+#define size 9
+using namespace std;
+int sum=0;
+bool fault[9][9]={false};
+
+
+class Sudoku
+{
+    int grid[size][size];
+    bool isSafe(int row, int col, int num); //To check whether the element is inserted in the right place
+    bool InRow(int row, int num);//To check whether element to be inserted is already present in the row or not
+    bool InBox(int boxStartRow, int boxStartCol, int num);//To check whether element to be inserted is already present in the box or not
+    bool InCol(int col, int num);//To check whether element to be inserted is already present in the column or not
+    public:
+        void clear_fault();
+        Sudoku(int b[9][9]); //To initialize the grid
+        bool FindUnassignedLocation(int &row, int &col); //Find a location in grid which is empty
+        bool SolveSudoku();//To solve the Sudoku
+        void replace_ele(int,int,int);//To replace the empty space by the number entered by user
+        void printGrid();//To print the required grid
+};
+
+void Sudoku::clear_fault()
+{
+    for (int i = 0; i < size; i++)
+        for ( int j= 0; j <size; j++)
+            if (fault[i][j])
+                grid[i][j]=0;

```

Comment:
Don't write code like that, enclose it as it might be ambiguous and error prone. Please, fix this!